core-services/src/main/java/org/zalando/nakadi/partitioning/PartitionResolver.java

```diff
@@ -66,9 +66,10 @@ public String resolvePartition(final EventType eventType, final JSONObject event
                     eventTypeStrategy);
         }
 
-        final List<String> partitions = timelineService.getTopicRepository(eventType)
-                .listPartitionNames(timelineService.getActiveTimeline(eventType).getTopic());
-        return partitionStrategy.calculatePartition(eventType, eventAsJson, partitions);
+        final int partitionsNumber = timelineService.getTopicRepository(eventType)
+                .getPartitionsNumber(timelineService.getActiveTimeline(eventType).getTopic());
```

Comment:
Here we are calling `getActiveTimeline()` two times (first inside of `getTopicRepository()`), and that one incurs taking a global lock.  For every published event.